unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDView.cs

```diff
@@ -1,39 +1,120 @@
+using System.Collections.Generic;
 using UnityEngine;
-using UnityEngine.Events;
 using UnityEngine.UI;
 
 public class TaskbarHUDView : MonoBehaviour
 {
     const string VIEW_PATH = "Taskbar";
 
-    public RectTransform windowContainer;
-    public Button chatButton;
-    public Button friendsButton;
+    [SerializeField] internal RectTransform windowContainer;
+    [SerializeField] internal CanvasGroup windowContainerCanvasGroup;
+    [SerializeField] internal LayoutGroup windowContainerLayout;
 
+    [SerializeField] internal TaskbarButton chatButton;
+    [SerializeField] internal TaskbarButton friendsButton;
+
+    [SerializeField] internal ChatHeadGroupView chatHeadsGroup;
+    internal List<TaskbarButton> taskbarButtonList = new List<TaskbarButton>();
     internal TaskbarHUDController controller;
 
-    internal static TaskbarHUDView Create(TaskbarHUDController controller)
+    internal void RefreshButtonList()
+    {
+        taskbarButtonList = new List<TaskbarButton>();
+        taskbarButtonList.Add(chatButton);
+        taskbarButtonList.Add(friendsButton);
+        taskbarButtonList.AddRange(chatHeadsGroup.chatHeads);
+    }
+
+    internal static TaskbarHUDView Create(TaskbarHUDController controller, IChatController chatController)
     {
         var view = Instantiate(Resources.Load<GameObject>(VIEW_PATH)).GetComponent<TaskbarHUDView>();
-        view.Initialize(controller);
+        view.Initialize(controller, chatController);
         return view;
     }
 
-    public void Initialize(TaskbarHUDController controller)
+    public void Initialize(TaskbarHUDController controller, IChatController chatController)
     {
         this.controller = controller;
+        chatHeadsGroup.Initialize(chatController);
+        chatButton.Initialize();
+        friendsButton.Initialize();
+
+        RefreshButtonList();
+
+        chatHeadsGroup.OnHeadToggleOn += OnWindowToggleOn;
+        chatHeadsGroup.OnHeadToggleOff += OnWindowToggleOff;
+
+        chatButton.OnToggleOn += OnWindowToggleOn;
+        chatButton.OnToggleOff += OnWindowToggleOff;
+
+        friendsButton.OnToggleOn += OnWindowToggleOn;
+        friendsButton.OnToggleOff += OnWindowToggleOff;
```

Comment:
I would unsubscribe all these subscriptions in the "OnDestroy" event.

Reply:
Not entirely necessary because those objects are inside the gameObject (i.e. will get destroyed with it) But I'll add them anyways for correctness sake.